Troubleshooting guide for the Vault Kubernetes KMS plugin (#1893)
hashicorp/web-unified-docs: content/vault/v1.21.x/content/docs/deploy/kubernetes/kms/troubleshooting.mdx

attached to the AppRole includes the required permissions:

```hcl
path "transit/keys/<KEY_NAME>" {
    capabilities = ["read"]
}

path "transit/encrypt/<KEY_NAME>" {
    capabilities = ["update"]
}

path "transit/decrypt/<KEY_NAME>" {
    capabilities = ["update"]
}

path "sys/license/status" {
    capabilities = ["read"]
}
```

This assumes your Transit secrets engine is mounted to the path `transit`. 
If you have chosen a different path for your Transit secrets engine, you will need to put that in the policy instead.

The `sys/license/status` path is required for Vault Enterprise validation. If you
omit this permission, the plugin cannot verify that you are running Vault Enterprise and will not start.

### Incorrect auth mount or AppRole name

If you see authentication errors in the plugin logs, verify the following:
- The `--approle-role-id` flag matches the Role ID for your AppRole in Vault.
- The `--approle-secret-id-path` flag references a valid file system path to a valid Secret ID for the AppRole.
- The static Pod running the plugin has access to the file system path referenced by `--approle-secret-id-path`.

> - The AppRole auth method is enabled at the expected mount path in Vault.

Reviewer: ```suggestion
- You enabled the AppRole auth method at the expected mount path in Vault.
```
Style correction: write in active voice

Also, what does "expected mount path" mean? Expected by what/whom? What should folks compare the mount path to in order to know if it's the "expected" path?

- The `--approle-role-id` flag matches the Role ID for your AppRole in Vault.
- The `--approle-secret-id-path` flag references a valid file system path to a valid Secret ID for the AppRole.
- The static Pod running the plugin has access to the file system path referenced by `--approle-secret-id-path`.
- You enabled the AppRole auth method in Vault at the same mount path that you provided to the plugin.

### Vault is unreachable

If you see connection errors in the plugin logs, verify the following:

- The `--vault-address` flag points to a reachable Vault server.
- Network policies or firewalls allow traffic from the control plane node to Vault.
- If you use TLS, make sure one of the following is true:
  - The `--vault-ca` flag points to a valid CA certificate.
  - The system trust store includes a valid CA for the Vault server.

### Expired AppRole Secret ID

If the plugin worked previously but authentication starts failing, the AppRole
Secret ID may have expired.

1. Generate a new Secret ID in Vault and replace the contents of the file at
   `--approle-secret-id-path`:

   ```shell-session
   $ vault write -f auth/approle/role/<role_name>/secret-id

> 

Reviewer: ````suggestion
   ```
````
Missing closing code fence

the plugin.

### Missing or invalid configuration flags

Read the plugin's logs in the static Pod with:
```shell-session
$ kubectl logs -n <NAMESPACE> <POD_NAME>
```

Look for validation errors at the start of the logs to see if a configuration flag or environment variable was 
passed incorrectly and causing the plugin to fail to start.

## Encryption or decryption errors

If encryption or decryption operations fail after the plugin connects to Vault,
the Transit key configuration may be incompatible.

### Key version constraints

If decryption fails for data encrypted with an older key version, verify the
`min_decryption_version` setting on the Transit key in Vault:

```shell-session
$ vault read transit/keys/<KEY_NAME>
```

If `min_decryption_version` is higher than the key version used to encrypt the data,
Vault rejects decryption requests. Lower the `min_decryption_version` or re-encrypt
the data with a newer key version.

~> **Warning:** Changing `min_decryption_version` can prevent Kubernetes from reading
data encrypted with older key versions. Verify that all encrypted data uses a key

> version at or above the new minimum before increasing this value.

Reviewer: ```suggestion
Changing `min_decryption_version` can prevent Kubernetes from reading data
encrypted with older key versions. Verify that all encrypted data uses a key
version at or above the new minimum before increasing the minimum version cutoff.
```
Style correction: avoid "this" as a pronoun

Also, there's no need to wrap this in a warning since it's a natural following paragraph for the previous one.
